### ISSUE TITLE
updateRepo() adds old versions: accidentally or on purpose?

### ISSUE THREAD

**Issue by eddelbuettel** [eddelbuettel/drat#99]
Hi @FelixErnst -- sorry for the slow movement on `drat` but some other packages are / were in the way.  So in preparation for launching this update code base (and thanks again for all the work!) I just gave `updateRepo(".")` a whirl on a `drat` I have here.

I moved the existing three `PACKAGE*` file to a 'hidden' subdirectory first, and then compared.  When running `updateRepo(".")` in isolation, it adds multiple versions of the same package which the base R function `tools::write_PACKAGES` does not do.  Now both `tools::write_PACKAGES` and `tools::update_PACKAGES` have `lastestOnly=TRUE` so I am confused how/why this happened.

And in the grand scheme of things, do we care enough to change or document?

**Comment by FelixErnst** [comment]
Hi @eddelbuettel,

I am looking forward to the tiledb package. Maybe that can help my squash some memory limits for `RNAmodR`

Regarding `drat` and `insertPackage`/`updateRepo` that was indeed intented. My intention was that the content of PACKAGES should always reflect the files present. Therefore `write_PACKAGES` runs with `latestOnly = FALSE` in `insertPackage` and `update_PACKAGES` should do the same.

I noticed a couple of things, after you posted, when looking into this in more detail:
1. For whatever reason I forgot to set `latestOnly = FALSE`  for `update_PACKAGES`. So I am a bit supprised that, you ended up with the multiple entries. Are your sure, that is not the result of a call to `insertPackage`?
2.  `latestOnly = FALSE`  is only used by `update_PACKAGES`, if there are new packages, not if packages are deleted. Because of this behaviour the test didn't catch the mistake made in 1.

If you would like to stick to base R behaviour that is also, fine for me. From my current point of view, it doesn't make that much sense, because the state becomes less defined, but once upon a time bandwidth was probably more of an issue.

I fixed the man pages, the "bug" and included additional tests.

**Comment by eddelbuettel** [comment]
> I am looking forward to the tiledb package.

In good shape at github and availble via `install.packages("tiledb", repos="https://ghrr.github.io")` "for now" while we await the grade of CRAN ...

As for "Are your sure" I think so.  I usually only ever call `dratInsert.r`, a minimal wrapper from littler.  But you are _correct_ in that I goofed: I misred old/new in the `diff` and my _existing_ `PACKAGES` already had _e.g._ `tiledb` 0.7.0 and 0.7.1.  So my bad, didn't know what I was looking at.  Gaaaah.
